### PR TITLE
docs: Updating globalfee docs

### PR DESCRIPTION
**Pull request by mpoke**

### CODE REVIEW

File: docs/modules/globalfee.md [hunk 1]
```diff
@@ -1,45 +1,106 @@
...
+## Bypass Fees Message Types
+
+Bypass messages are messages that are exempt from paying fees. The above global fee and `minimum-gas-prices` checks do not apply to bypass message types if these two conditions are met:
+- the transaction list contains only bypass message types (bypass messages and non-bypass messages cannot be combined in a single transaction list)
+- the total gas of used is less than or equal to `len(messages) * MaxBypassMinFeeMsgGasUsage` (Please note: the current `MaxBypassMinFeeMsgGasUsage` is set to 200,000).
+- the transaction fee denom has to be a subset of denoms defined by the global fee
+
+Node operators can configure `bypass-min-fee-msg-types` in `config/app.toml`.
+
+Nodes inited by Gaiad `v7.0.2` or later use `["/ibc.core.channel.v1.MsgRecvPacket", "/ibc.core.channel.v1.MsgAcknowledgement","/ibc.applications.transfer.v1.MsgTransfer"]` as defaults. Nodes with `bypass-min-fee-msg-types = []` or missing this field in `app.toml` also use default bypass message types.
```
Comment (mmulji-ic) [inline docs/modules/globalfee.md]: inited?

File: docs/modules/globalfee.md [hunk 2]
```diff
@@ -1,45 +1,106 @@
...
+
+Nodes created using Gaiad `v7.0.1` or earlier do not have `bypass-min-fee-msg-types` configured in `config/app.toml` - they are alsousing default values. The `bypass-min-fee-msg-types` config option can be added to `config/app.toml` before the `[telemetry]` field.
```
Comment (mmulji-ic) [inline docs/modules/globalfee.md]: alsousing -> add space

File: ante/ante.go [hunk 3]
```diff
@@ -46,14 +46,20 @@ func NewAnteHandler(opts HandlerOptions) (sdk.AnteHandler, error) {
 		sigGasConsumer = ante.DefaultSigVerificationGasConsumer
 	}
 
+	// maxBypassMinFeeMsgGasUsage is the maximum gas usage per message
+	// so that a transaction that contains only message types that can
+	// bypass the minimum fee can be accepted with a zero fee.
+	// For details, see gaiafeeante.NewFeeDecorator()
+	var maxBypassMinFeeMsgGasUsage uint64 = 200_000
```
Comment (MSalopek) [inline ante/ante.go:53]: Nit: consider making this a const and defining it at the top of the file for clarity.

File: docs/modules/globalfee.md [hunk 4]
```diff
@@ -1,45 +1,100 @@
...
-- global fees (are defined at the network level, via [Gov Proposals](https://hub.cosmos.network/main/governance/proposals/))
-- minimum-gas-prices (are specified validator node, in the `config/app.toml` configuration file)
-  
-### Global Fees
+2. `minimum-gas-prices`
+This parameter is part of the node configuration, it can be set in the `config/app.toml` configuration file.
+
+3. `bypass-min-fee-msg-types`
+This parameter is part of the node configuration, it can be set in the `config/app.toml` configuration file.
+This represent a list of message types that will be excluded from paying any fees for inclusion in a block.
```
Comment (yaruwangway) [inline docs/modules/globalfee.md:14]: represent -> represents

File: x/globalfee/ante/fee.go [hunk 5]
```diff
@@ -59,11 +59,12 @@ func (mfd FeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, ne
...
-	// Only check for minimum fees and global fee if the execution mode is CheckTx and the tx does
-	// not contain operator configured bypass messages. If the tx does contain
-	// operator configured bypass messages only, it's total gas must be less than
-	// or equal to a constant, otherwise minimum fees and global fees are checked to prevent spam.
-	// please note bypassing fees requires 1. all the msgs are bypass msgs types, 2. total gas of bypass msgs should be <= len(msgs)*mfd.MaxBypassMinFeeMsgGasUsage (presently MaxBypassMinFeeMsgGasUsage is 200,000). Please check ../../../docs/modules/globalfee.md for more info.
+	// Accept zero fee transactions only if both of the following statements are true:
+	// 	- the tx contains only message types that can bypass the minimum fee,
+	//	see BypassMinFeeMsgTypes;
+	//	- the total gas limit per message does not exceed MaxBypassMinFeeMsgGasUsage,
+	//	i.e., <= len(msgs) * MaxBypassMinFeeMsgGasUsage
```
Comment (MSalopek) [inline x/globalfee/ante/fee.go]:
```suggestion
	//	i.e., totalGas <= len(msgs) * MaxBypassMinFeeMsgGasUsage
```

**Review by MSalopek — APPROVED** [review]
LGTM! Left nits.

File: ante/ante.go (hunk 3 above)
Comment (mpoke) [reply ante/ante.go:53]: We want to refactor the entire thing anyway, so I don't think it's important

File: docs/modules/globalfee.md [hunk 6]
```diff
@@ -1,45 +1,100 @@
...
 - fees must have non-negative amount, with a valid and unique denom (i.e. no duplicate denoms are allowed).
 
-Global fees allow denoms with zero coins. Zero value coins are used to define fee denoms, in cases where a fee payment is made voluntarily, even if the chain does not enforce fees, zero-value coins can be used to define the fee denoms this chain accepts.
 
-Each transaction (except bypass transactions) have to meet the following global fee requirements:
+There are **two exceptions** from the global fees rules that allow zero fee transactions:
+
+1. Transactions that contain only [message types that can bypass the minimum fee](#bypass-fees-message-types) may have zero fees. We refer to this as _bypass transactions_. Node operators can choose to define these message types (for each node) via the `bypass-fee-message-types` configuration parameter.
+
+2. If one of the entries in the global fees list has a zero amount, e.g., `0uatom`, and the corresponding denom, e.g., `uatom`, is not present in `minimum-gas-prices`.
```
Comment (yaruwangway) [inline docs/modules/globalfee.md:37]:
```suggestion
2. One of the entries in the global fees list has a zero amount, e.g., `0uatom`, and the corresponding denom, e.g., `uatom`, is not present in `minimum-gas-prices`.
```

**Review by yaruwangway — APPROVED** [review]

**Pull request merged**